payment-backend-service/src/main/java/backend/service/service/RetryManager.java

```diff
@@ -0,0 +1,30 @@
+package backend.service.service;
+
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import io.github.resilience4j.retry.RetryRegistry;
+import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+@Component
+@NoArgsConstructor
+public class RetryManager {
+
+    public <T>  T retryRequest(int maxAttempts, int waitDuration, Predicate<T> condition, Supplier<T> action){
```

Comment:
We have to avoid cases when we need more than 2 arguments for method. You can use simple structure for it.